pxtlib/util.ts

```diff
@@ -1948,6 +1948,62 @@ namespace ts.pxtc.Util {
 
         return enabled;
     }
+
+    /**
+     * Remove potentially sensitive info from the given data to avoid logging it.
+     * Currently only supports string data.
+     */
+    export function cleanData(data: string): string;
+    export function cleanData(data: pxt.Map<string | number>): pxt.Map<string | number>;
+    export function cleanData(data: any): any {
+        if (!data) return data;
+
+        let result: any;
+        if (typeof data === "string") {
+            result = removePropertiesWithPossibleUserInfo(data);
+        } else if (typeof data === "object") {
+            result = {};
+            for (const [key, value] of Object.entries(data)) {
+                result[key] = typeof value === "string" ? removePropertiesWithPossibleUserInfo(value) : value;
+            }
+        } else {
+            result = data;
+        }
+
+        return result;
+    }
+
+    /**
+     * Attempts to remove commonly leaked PII
+     * @param property The property which will be removed if it contains user data
+     * @returns The new value for the property
+     * 
+     * Taken from https://github.com/microsoft/vscode/blob/main/src/vs/platform/telemetry/common/telemetryUtils.ts
+     */
+    function removePropertiesWithPossibleUserInfo(property: string): string {
+        // If for some reason it is undefined we skip it (this shouldn't be possible);
+        if (!property) {
+            return property;
+        }
+
+        const userDataRegexes = [
+            { label: 'Google API Key', regex: /AIza[A-Za-z0-9_\\\-]{35}/ },
+            { label: 'Slack Token', regex: /xox[pbar]\-[A-Za-z0-9]/ },
+            { label: 'GitHub Token', regex: /(gh[psuro]_[a-zA-Z0-9]{36}|github_pat_[a-zA-Z0-9]{22}_[a-zA-Z0-9]{59})/ },
+            { label: 'Generic Secret', regex: /(key|token|sig|secret|signature|password|passwd|pwd|android:value)[^a-zA-Z0-9]/i },
```

Comment:
Suggest putting a few guardrails on this one:
- `\b` - don't match mid-word. You may get false positives matching "key" and "sig".
- `{0,200}` don't scan more than 200 chars.
```suggestion
            { label: 'Generic Secret', regex: /\b(key|token|sig|secret|signature|password|passwd|pwd|android:value)[^a-zA-Z0-9]{0,200})/i },
```